test/Tests.csproj

```diff
@@ -28,22 +28,28 @@
     </Content>
   </ItemGroup>
 
-
-
   <ItemGroup>
+    <PackageReference Include="coverlet.collector" Version="6.0.1">
+      <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
+      <PrivateAssets>all</PrivateAssets>
+    </PackageReference>
+    <PackageReference Include="coverlet.msbuild" Version="6.0.1">
+      <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
+      <PrivateAssets>all</PrivateAssets>
+    </PackageReference>
     <PackageReference Include="Microsoft.NET.Test.Sdk" Version="17.8.0" />
     <PackageReference Include="xunit" Version="2.6.2" />
     <PackageReference Include="xunit.runner.visualstudio" Version="2.5.4" PrivateAssets="all" />
-    <PackageReference Include="NSubstitute" Version="5.1.0" />
+    <PackageReference Include="Moq" Version="4.20.70" />
```

Comment:
Why the NSubstitute --> Moq switch?

Reply:
This is an easy one and partially opinionated. I'm not going to go into a huge list of pros and cons... but somethings are...

NSub relies on more static state in some cases... this means failures are more likely missed and will happen through the call stack and into other tests - i.e. encapsulation can be an issue where an issue in one test can manifest as errors in other tests. I prefer the option with better state management.

Moq is one of the oldest, most maintained, and more poplar mocking testing libs.

Also, with Moq you can mock  protected members and non-virtual methods (with some limitations) - much like what we were chatting about the other day.

I personally prefer the fluent and expressive syntax - so the lambda expressions and semantic API chains just read better IMO - so instead of:

```
var thingServiceMock = Substitute.For<IThingService>();
thingServiceMock.Received().SendThing(Arg.Any<Thing>());
```

there is this in Moq

```
var thingServiceMock = new Mock<IQuoteService>();
thingServiceMock.Verify(q => q.SendThing(It.IsAny<Thing>()));
```

While this is purely subjective, I feel like these things lead to more readable code.

The reason I used NSubstitute initially was because it was the pattern used in Octokit.net and I assumed that it would be a more simplistic transition.

Let me know your thoughts.